paymentapi/src/test/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContextBuilderUnitTests.java

```diff
@@ -0,0 +1,147 @@
+
+        assertEquals("update/url1", context.getUpdateCardRedirectSuccessUrl());
+        assertEquals("update/url2", context.getUpdateCardRedirectCancelUrl());
+        assertEquals("update/url3", context.getUpdateCardCallbackSuccessUrl());
+        assertEquals("update/url4", context.getUpdateCardCallbackCancelUrl());
+    }
+
+    @Test
+    public void testBuildContextWithMissingPaytrailMerchantId() throws Exception {
+        Mockito.when(commonServiceConfigurationClient.getMerchantModel(any(), any()))
+                .thenReturn(createMockMerchantDto(NAMESPACE, MERCHANT_ID, null));
+        Mockito.when(commonServiceConfigurationClient.getMerchantPaytrailSecretKey(any())).thenReturn(PAYTRAIL_SECRET_KEY);
+
+        Exception exception = assertThrows(Exception.class, () -> {
+            paymentContextBuilder.buildFor(NAMESPACE, MERCHANT_ID, false);
+        });
+        CommonApiException cae = (CommonApiException) exception;
+        assertEquals(CommonApiException.class, cae.getClass());
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, cae.getStatus());
+        assertEquals("failed-to-get-paytrail-merchant-id", cae.getErrors().getErrors().get(0).getCode());
+    }
+
+    @Test
+    public void testBuildContextWithMissingPaytrailSecret() throws Exception {
+        Mockito.when(commonServiceConfigurationClient.getMerchantModel(any(), any()))
+                .thenReturn(createMockMerchantDto(NAMESPACE, MERCHANT_ID, PAYTRAIL_MERCHANT_ID));
+        Mockito.when(commonServiceConfigurationClient.getMerchantPaytrailSecretKey(any())).thenReturn(null);
+
+        Exception exception = assertThrows(Exception.class, () -> {
+            paymentContextBuilder.buildFor(NAMESPACE, MERCHANT_ID, false);
+        });
+        CommonApiException cae = (CommonApiException) exception;
+        assertEquals(CommonApiException.class, cae.getClass());
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, cae.getStatus());
+        assertEquals("failed-to-get-merchant-paytrail-secret-key", cae.getErrors().getErrors().get(0).getCode());
+    }
+
+    private MerchantDto createMockMerchantDto(String namespace, String merchantId, String paytrailMerchantId) {
+        MerchantDto mockMerchantDto = new MerchantDto();
+        mockMerchantDto.setNamespace(namespace);
+        mockMerchantDto.setMerchantId(merchantId);
+        mockMerchantDto.setCreatedAt(LocalDateTime.now());
+        mockMerchantDto.setUpdatedAt(LocalDateTime.now());
+
+        ConfigurationDto mockConfigurationDto = new ConfigurationDto();
+        mockConfigurationDto.setKey(ServiceConfigurationKeys.MERCHANT_PAYTRAIL_MERCHANT_ID);
+        mockConfigurationDto.setValue(paytrailMerchantId);
+
+        ArrayList<ConfigurationDto> mockConfigurations = new ArrayList<>();
+        mockConfigurations.add(mockConfigurationDto);
+        mockMerchantDto.setConfigurations(mockConfigurations);
+
+        return mockMerchantDto;
```

Comment:
Could you add delete for this mockMerchant when test is run

Reply:
It is not created into database. It is used for mocked call to  commonServiceConfigurationClient.getMerchantModel in https://github.com/City-of-Helsinki/verkkokauppa-core/blob/3234d3bebc156d5fb0622724ff283372ff0cffe9/paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContextBuilder.java#L64